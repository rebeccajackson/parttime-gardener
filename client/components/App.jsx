import React from 'react'
import {HashRouter as Router} from 'react-router-dom'

import Home from './Home'

const App = () => {  
  return(
  
    <div id="home">
      <Home/>
    </div>

  )
}

export default App

