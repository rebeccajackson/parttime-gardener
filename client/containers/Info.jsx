import React, {Fragment} from 'react'
import {connect} from 'react-redux'

import {addToGarden} from '../api/users'

class Info extends React.Component{

  addToGarden = () => {
    const payload = {user: this.props.user, veg: this.props.currentVeg}
    const {dispatch} = this.props
    dispatch(addToGarden(payload))
  }


  render(){
    const {currentVeg} = this.props
    
    return Object.keys(currentVeg).length == 0 ? (
      <Fragment>
        <img className="icon" src="/images/information.png" alt=""/>
        <div className='info-box'>
        <br/>
        <h3>Click on a veg</h3>
        </div>
      </Fragment>
    ) :
      <Fragment>
        <img className="icon" src="/images/information.png" alt=""/>
        <div className='info-box'>
          <br/>
          <h2>{currentVeg.name}</h2>
          <br/>
          <h4>Tips</h4>
          <p>{currentVeg.tips}</p>
          <br/>
          <h4>Plant spacing</h4>
          <p>{currentVeg.spaceplants}</p>
          <button className='button' onClick={this.addToGarden}>
            Add to garden
          </button>
        </div>
      </Fragment>
  }
}
function mapStateToProps({user, currentVeg}){
  return {user, currentVeg}
}

export default connect(mapStateToProps)(Info)
