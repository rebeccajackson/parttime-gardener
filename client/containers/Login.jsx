import React from 'react'
import {connect} from 'react-redux'
import {loginUser, loginError} from '../actions/auth/login'

class Login extends React.Component {
  constructor(props) {
    super(props)
    this.state = {
      username: '',
      password: ''
    }
    this.updateDetails = this.updateDetails.bind(this)
    this.submit = this.submit.bind(this)
  }
  componentDidMount() {
    this.props.dispatch(loginError(''))
  }
  updateDetails(e) {
    this.setState({[e.target.name]: e.target.value})
  }
  submit(e) {
    e.preventDefault()
    let {username, password} = this.state
    this.props.dispatch(loginUser({username, password}))
  }
  render() {
    const {auth} = this.props
    return (
      <form className="form box" onSubmit={this.submit}>
        <h1 className="">Login</h1>
        <hr />
        {auth.errorMessage && <span className="">{auth.errorMessage}</span>}
        <label className="">Username
          <input required className="" placeholder="User Name" type="text" name="username" onChange={this.updateDetails}/>
        </label>
        <label className="">Password
          <input required className="" placeholder="Password" type="password" name="password" onChange={this.updateDetails}/>
        </label>
        <input className="" value='Login' type="submit" />
      </form>
    )
  }
}

const mapStateToProps = ({auth}) => {
  return {
    auth
  }
}

export default connect(mapStateToProps)(Login)
