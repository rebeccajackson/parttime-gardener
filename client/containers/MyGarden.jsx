import React from 'react'
import {HashRouter as Router, Link} from 'react-router-dom'
import { connect } from 'react-redux'

import { getUserVeges } from '../api/vegs';

class MyGarden extends React.Component{

  componentDidMount(){
    const { dispatch } = this.props
    dispatch(getUserVeges(this.props.user))
  }

  render(){
    const { user, userVeges } = this.props
    return(
      <Router >
        <div className="my-garden">
          <h3 className="box-title">{user.name}'s garden</h3>

          <div className="garden-list">
            {userVeges.map((obj, i) => 

            <Link className='vegNameList' key={i} 
              onClick={this.props.handleVegClick.bind(this, obj)}
              to={ `/veg/${obj.name}`}>
              {obj.name}
            </Link>
            )}
          </div>
        </div>
      </Router>
    )
  }
}

function mapStateToProps({user, userVeges}){
  return ({ user, userVeges })
}

export default connect(mapStateToProps)(MyGarden)