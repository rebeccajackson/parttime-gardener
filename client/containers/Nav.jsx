import React from 'react'
import {connect} from 'react-redux'
import {Link} from 'react-router-dom'
import {logoutUser} from '../actions/auth/logout'

class Nav extends React.Component {
  constructor(props) {
    super(props)
    this.state = {
      showBurger: false
    }
    this.toggleBurger = this.toggleBurger.bind(this)
  }
  toggleBurger() {
    this.setState({showBurger: !this.state.showBurger})
  }
  render() {
    const {auth, logout} = this.props
    const {showBurger} = this.state
    return (
      <nav className="navbar">
        <div className="container">
          <div className="navbar-brand">
            <span onClick={this.toggleBurger}
              className={`navbar-burger burger ${showBurger ? 'is-active': ''}`} 
              data-target="navbarMenuHeroA">
              <span></span>
              <span></span>
              <span></span>
            </span>
          </div>

          <div id="navbarMenuHeroA" className={`navbar-menu ${showBurger ? "is-active" : ''}`}>
            <div className="navbar-end">
              {auth.isAuthenticated ?
                [ <Link to='/' className="navbar-item" onClick={() => logout()}>
                    Logout
                  </Link> ] 
                : 
                [ <Link onClick={this.toggleBurger} className="navbar-item is-large" to='/login'>
                    Login
                  </Link>,
                  <Link onClick={this.toggleBurger} className="navbar-item" to='/register'>
                    Register
                  </Link> ]
                
              }
            </div>
          </div>
        </div>
      </nav>
    )
  }
}

const mapDispatchToProps = (dispatch) => {
  return {
    logout: () => dispatch(logoutUser())
  }
}

const mapStateToProps = ({auth}) => {
  return {auth}
}

export default connect(mapStateToProps, mapDispatchToProps)(Nav)
