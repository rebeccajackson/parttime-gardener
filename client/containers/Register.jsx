import React, { Component } from 'react'

import { connect } from 'react-redux'
import { registerUser } from '../actions/auth/register'

class Register extends Component {
  constructor(props){
    super(props)
    this.state = {
      user: {}
    }
    this.handleChange = this.handleChange.bind(this)
    this.handleSubmit = this.handleSubmit.bind(this)
  } 

  
  handleChange(e){
    let user = this.state.user
    user[e.target.name] = e.target.value
    this.setState({ user })
  }
  

  handleSubmit(e) {
    e.preventDefault();
    const {dispatch} = this.props
    const creds = this.state.user
    dispatch(registerUser(creds))
  }
  render() {
    return (
      <div className="home-page">
          <form className="form" onSubmit={this.handleSubmit}>
          
              <input className='form-input' type="text" name='username'  onChange={this.handleChange} />
              <input className='form-input' type="password" name='password' onChange={this.handleChange} />
        
              <input className="button" type="submit" value="Login" />
          </form>
        </div>
    )
  }
}



export default connect()(Register)