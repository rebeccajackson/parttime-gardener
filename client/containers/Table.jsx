import React, {Fragment} from 'react'
import {HashRouter as Router} from 'react-router-dom'
import { connect } from 'react-redux'


import PlantingMonth from '../components/PlantingMonth';
import ViewMonth from '../containers/ViewMonth'
import ViewVeg from '../containers/ViewVeg'
import MyGarden from '../containers/MyGarden'

import { getMonths } from '../api/months'
import { getMonthVeges } from '../api/months'
import {displayViewVeg} from '../actions/index'
import {getPlantingMonthsArr} from '../api/vegs'

class Table extends React.Component{
  constructor(props){
    super(props)
    this.state = {
      vegClick: false,
      monthClick: false
    }
  }

  componentDidMount(){
    const { dispatch } = this.props
    dispatch(getMonths())
  }

  handleVegClick = (veg) =>{
    const { dispatch } = this.props
    dispatch(displayViewVeg(veg))
    this.setState({
      monthClick: false,
      vegClick: true
    })
    dispatch(getPlantingMonthsArr(veg))
  }

  handleMonthClick = (month) => {
    const { dispatch } = this.props
    dispatch(getMonthVeges(month))
    this.setState({
      monthClick: true,
      vegClick: false
    })
  }

  render(){
    const { months, plantingMonths } = this.props
    return(
      <Router>
          <Fragment>
            <div className='table'>
              <div className='header-grid'>
                <div className='grid12'>
                  {months.map((month, i) => 
                    <button onClick={this.handleMonthClick.bind(this, month)} 
                        className='month-letter overlay heartbeat' key={i}
                        >
                      
                        <div className={`${month.season} table-header`}>
                          {month.name.charAt(0)}
                        </div>
                    </button> 
                  )}
                </div>  
              </div>
    
              <div className='plantingMonths'> 
                <div className='grid12'>
                  {plantingMonths.map((month, i) => 
                    <div key={i} className={`${month.name} plantingMonths-header`}>
                      <PlantingMonth show={month.show}/>
                    </div>
                  )}
              </div> 
            </div>
          </div>
           <div className='contents'>
            <MyGarden handleVegClick={this.handleVegClick}/>
           {this.state.monthClick && <ViewMonth />
              || this.state.vegClick && <ViewVeg />
              }
          </div>
        </Fragment>
      </Router>
    )
  }
}

function mapStateToProps({ months, plantingMonths }) {
  return { months, plantingMonths }
}

export default connect(mapStateToProps)(Table)