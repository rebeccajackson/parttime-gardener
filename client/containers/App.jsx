import React, {Fragment} from 'react'
import {HashRouter as Router, Link} from 'react-router-dom'
import { connect } from 'react-redux'

import Table from './Table'
import Register from './Register'

import { login } from '../api/users';


class App extends React.Component{
  constructor(props){
    super(props)
    this.state = {
      user: {}
    }
    this.handleChange = this.handleChange.bind(this)
    this.handleLogin = this.handleLogin.bind(this)
  } 

  
  handleChange(e){
    let user = this.state.user
    user[e.target.name] = e.target.value
    this.setState({ user })
  }

  handleLogin(e) {
    e.preventDefault();
    const {dispatch} = this.props
    const user = this.state.user
    dispatch(login(user))
  }

  render(){
    return(
      <Router>
        <Fragment>
          <div id="home">
            <div className="home-page">
              <form className="form" onSubmit={this.handleLogin}>
              
                  <input className='form-input' type="text" name='username'  onChange={this.handleChange} />
                  <input className='form-input' type="password" name='password' onChange={this.handleChange} />
            
                  <input className="button" type="submit" value="Login" />
              </form>
              <div>
                <Link to='/register'>Signup</Link>
              </div>
            </div>
            {this.props.userLogin.login && 
              <Table />
            }        
          </div>
        </Fragment>
      </Router>

    )
  }
}

function mapStateToProps({ userLogin}) {
  return { userLogin }
}

export default connect(mapStateToProps)(App)

