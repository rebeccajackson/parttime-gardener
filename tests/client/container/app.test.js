import React from 'react'
import { App } from '../../../client/containers/App'

import { shallow} from 'enzyme'


describe('App container', () => {

  it('displays Table container if isAuthenticated is true', () => {
    const wrapper = shallow(<App auth={{isAuthenticated: true}}/>)
    const actual = wrapper.find('Connect(Table)').length

    expect(actual).toEqual(1)
  })

  it('renders div.home-page when isAuthenticated is false', () => {
    const wrapper = shallow(<App auth={{isAuthenticated: false}}/>)
    const expected = 1
    const actual = wrapper.find('.home-page').length

    expect(actual).toBe(expected)
  })

  it('is the same snapshot', () => {
    const wrapper = shallow(<App auth={{isAuthenticated: false}}/>)
    expect(wrapper).toMatchSnapshot()

  })
})
