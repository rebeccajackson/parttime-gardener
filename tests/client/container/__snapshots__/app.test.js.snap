// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App container is the same snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    auth={
      Object {
        "isAuthenticated": false,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <React.Fragment>
        <div
          id="home"
        >
          <Connect(Nav) />
          <div
            className="home-page"
          >
            <Route
              component={[Function]}
              path="/login"
            />
            }
            <Route
              component={[Function]}
              path="/register"
            />
          </div>
        </div>
      </React.Fragment>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <div
          id="home"
        >
          <Connect(Nav) />
          <div
            className="home-page"
          >
            <Route
              component={[Function]}
              path="/login"
            />
            }
            <Route
              component={[Function]}
              path="/register"
            />
          </div>
        </div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Connect(Nav) />,
            <div
              className="home-page"
            >
              <Route
                component={[Function]}
                path="/login"
              />
              }
              <Route
                component={[Function]}
                path="/register"
              />
            </div>,
          ],
          "id": "home",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <Route
                  component={[Function]}
                  path="/login"
                />,
                "}",
                <Route
                  component={[Function]}
                  path="/register"
                />,
              ],
              "className": "home-page",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "path": "/login",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "}",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "path": "/register",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": Symbol(react.fragment),
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <React.Fragment>
          <div
            id="home"
          >
            <Connect(Nav) />
            <div
              className="home-page"
            >
              <Route
                component={[Function]}
                path="/login"
              />
              }
              <Route
                component={[Function]}
                path="/register"
              />
            </div>
          </div>
        </React.Fragment>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <div
            id="home"
          >
            <Connect(Nav) />
            <div
              className="home-page"
            >
              <Route
                component={[Function]}
                path="/login"
              />
              }
              <Route
                component={[Function]}
                path="/register"
              />
            </div>
          </div>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Connect(Nav) />,
              <div
                className="home-page"
              >
                <Route
                  component={[Function]}
                  path="/login"
                />
                }
                <Route
                  component={[Function]}
                  path="/register"
                />
              </div>,
            ],
            "id": "home",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <Route
                    component={[Function]}
                    path="/login"
                  />,
                  "}",
                  <Route
                    component={[Function]}
                    path="/register"
                  />,
                ],
                "className": "home-page",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "component": [Function],
                    "path": "/login",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "}",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "component": [Function],
                    "path": "/register",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
        "type": Symbol(react.fragment),
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
